sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -664,3 +664,24 @@ def test_lasso_non_float_y():
         clf_float = model(fit_intercept=False)
         clf_float.fit(X, y_float)
         assert_array_equal(clf.coef_, clf_float.coef_)
+
+
+def test_enet_float_precision():
+    # Generate dataset
+    X, y, X_test, y_test = build_dataset(n_samples=20, n_features=10)
+    # Here we have a small number of iterations, and thus the
+    # ElasticNet might not converge. This is to speed up tests
+
+    for fit_intercept in [True, False]:
+        coef = {}
+        clf = ElasticNet(alpha=0.5, max_iter=100, precompute=False,
+                         fit_intercept=fit_intercept)
+        for dtype in [np.float64, np.float32]:
+            X = dtype(X)
+            y = dtype(y)
+            ignore_warnings(clf.fit)(X, y)
+
+            coef[dtype] = clf.coef_
+
+        assert_array_almost_equal(coef[np.float32], coef[np.float64],
+                                  decimal=4)
```

Comment:
should be checking `intercept_` too. and the dtypes of the `coef_` if appropriate.